drivers/usb/udc/udc_stm32.c

```diff
@@ -768,6 +775,21 @@ static int udc_stm32_ep_dequeue(const struct device *dev,
 	return 0;
 }
 
+static enum udc_bus_speed udc_stm32_device_speed(const struct device *dev)
+{
+	struct udc_stm32_data *priv = udc_get_private(dev);
+
+	if (priv->pcd.Init.speed == USBD_HS_SPEED) {
+		return UDC_BUS_SPEED_HS;
```

Comment:
looks like these constant are not defined in the HAL for F1 series mcus, should this go under an ifdef?

Reply:
They are defined in `include/zephyr/drivers/usb/udc.h`?

Reply:
sorry my bad, wrong line, yes

Reply:
Opened https://github.com/zephyrproject-rtos/zephyr/pull/75306

Reply:
> looks like these constant are not defined in the HAL for F1 series mcus, should this go under an ifdef?

I did not expect that, they define USBD_FS_SPEED for F1 but not USBD_HS_SPEED. @fabiobaltieri do you have a PR ready or should I fix it?

Reply:
yeah, guess the new stack needs a bit more CI build coverage, those HALs are evil they have all sort of inconsistencies